client/state/sites/reducer.js

```diff
@@ -100,13 +106,24 @@ export function items( state = {}, action ) {
 				return memo;
 			}, initialNextState );
 
-		case SITE_DELETE_RECEIVE:
+		case SITE_DELETE_RECEIVE: {
+			if ( state === null ) {
+				return state;
+			}
 			return omit( state, action.siteId );
+		}
```

Comment:
just for fun, we can also express this as a ternary expression instead of a conditional statement:

```js
case SITE_DELETE_RECEIVE:
	return state !== null
		? omit( state, action.siteId )
		: state;
```

Also, why are we creating a block via the curlies `{` and `}` on these `case` statements?